packages/wallet/src/Wallets/BaseWallet.ts

```diff
@@ -878,15 +878,33 @@ export class BaseWallet implements ObservableWallet {
     return isEmpty ? [knownAddresses[0]] : [];
   }
 
-  /** Update the witness of a transaction with witness provided by this wallet */
-  async updateWitness({ tx, sender }: UpdateWitnessProps): Promise<Cardano.Tx> {
-    return this.finalizeTx({
-      auxiliaryData: tx.auxiliaryData,
+  async addSignatures({ tx, sender }: AddSignaturesProps): Promise<Serialization.TxCBOR> {
+    const serializableTx = Serialization.Transaction.fromCbor(tx);
+    const auxiliaryData = serializableTx.auxiliaryData()?.toCore();
+    const body = serializableTx.body().toCore();
+    const hash = serializableTx.getId();
+    const witness = serializableTx.witnessSet().toCore();
+    const bodyCbor = serializableTx.body().toCbor();
+
+    const witnessedTx = await this.finalizeTx({
+      auxiliaryData,
+      bodyCbor,
       signingContext: {
         sender
       },
-      tx: { body: tx.body, hash: tx.id },
-      witness: tx.witness
+      tx: { body, hash },
+      witness
     });
+
+    const coreWitness = witnessedTx.witness;
+    const witnessSet = serializableTx.witnessSet();
+
+    witnessSet.setVkeys(
+      Serialization.CborSet.fromCore([...coreWitness.signatures], Serialization.VkeyWitness.fromCore)
```

Comment:
Is this preserving original signatures? :thinking: 
```suggestion
    witnessSet.setVkeys(
      Serialization.CborSet.fromCore([...coreWitness.signatures, ...witnessSet.vkeys().toCore()], Serialization.VkeyWitness.fromCore)
```

Reply:
Yes finalize preserves the original signatures, so coreWitness already has all the signatures